src/clients/base.rs

```diff
@@ -295,18 +295,28 @@ where
     }
 
     /// The manually paginated version of [`Self::artist_albums`].
-    async fn artist_albums_manual(
+    async fn artist_albums_manual<'a>(
         &self,
         artist_id: ArtistId<'_>,
-        album_type: Option<AlbumType>,
+        include_groups: impl IntoIterator<Item = AlbumType> + Send + 'a,
         market: Option<Market>,
         limit: Option<u32>,
         offset: Option<u32>,
     ) -> ClientResult<Page<SimplifiedAlbum>> {
         let limit = limit.map(|x| x.to_string());
         let offset = offset.map(|x| x.to_string());
+        let include_groups = include_groups
+            .into_iter()
+            .map(|t| t.into())
+            .collect::<Vec<&'static str>>();
+        let include_groups = include_groups
```

Comment:
Probably, It is more readable to have two different variables instead of shadowing one variable over the other.